### ISSUE TITLE
Consistency: Verify checkpoint with TUF metadata

### ISSUE THREAD

**Issue by haydentherapper** [sigstore/rekor-monitor#51]
**Description**

Currently the Rekor public key, needed to verify checkpoint signatures, is fetched from Rekor. This is not ideal, and we should use the public key in the TUF targets to verify checkpoints. This will mean we need to bundle the TUF root.

**Comment by linus-sun** [comment]
This also applies [here](https://github.com/sigstore/rekor-monitor/blob/48375db9da425b4d9985ca13cb9f854b4f59439d/pkg/ct/consistency.go#L43)- the CTFE public key should be fetched from TUF.

**Comment by facutuesca** [comment]
> This will mean we need to bundle the TUF root.

@haydentherapper if we were to get the TUF targets using `sigstore-go`'s TUF [client](https://github.com/sigstore/sigstore-go/tree/main/pkg/tuf), do we need to bundle the roots in this repo? I ask since `sigstore-go` already [bundles them](https://github.com/sigstore/sigstore-go/tree/main/pkg/tuf/repository) and you can create a client with either Prod or Staging roots:

https://github.com/sigstore/sigstore-go/blob/7212b45515a1387f7c8dd147747838355ce72427/pkg/tuf/options.go#L138-L141

**Comment by haydentherapper** [comment]
@facutuesca No, for the public instance, we can leverage sigstore-go directly and don't need to bundle the roots. We can add flags to provide your own root and and TUF repo and/or public key if you're not monitoring the public instance.

**Comment by facutuesca** [comment]
@haydentherapper I think this one can be closed
